plugin-server/src/worker/ingestion/PoE.md

```diff
@@ -0,0 +1,107 @@
+
+### 1. Filtering on person properties
+
+Latest person properties are joined in during query time.
+
+### 2. Insights counting unique persons
+
+Distinct_id to Person mapping is joined together at query time, see [docs](https://posthog.com/docs/how-posthog-works/queries#insights-counting-unique-persons).
+
+## PoEv1
+
+### 1. Filtering on person properties
+
+Properties from ingestion time are used, see [docs](https://posthog.com/docs/how-posthog-works/queries#filtering-on-person-properties)
+
+### 2. Insights counting unique persons
+
+Person IDs from ingestion time are used. Let's look at the same example from the [docs](https://posthog.com/docs/how-posthog-works/queries#insights-counting-unique-persons)
+
+| ID  | Event       | `person_id` |
+| --- | ----------- | ----------- |
+| 1   | viewed page | `user1`     |
+| 2   | viewed page | `user2`     |
+| 3   | viewed page | `user1`     |
+
+> **Note:** This isn't _exactly_ how `person_id`'s are stored within the events table, but it will help us to keep things simple.
+
+In this case, if we ran a query asking for the number of unique users who viewed a page, we would get a result of `2`, as our table contains 2 unique `person_id`'s.
+
+The way we write the `person_id` to each event has some implications for the number of unique users that are displayed:
+
+1. Some users are counted twice on the trend graph.
+   The source of truth for data is the events table. Since this is point-in-time data, it is not possible to determine whether two `person_id`'s were later merged into a single user, which results in them being counted separately.
+2. In the person modal, the count may be lower than the count displayed in the graph.
+   Persons who've been merged into one have one of their old IDs deleted. We remove these people from the persons modal, as there's no place to link them to.
+
+To understand better how these scenarios can arise, let's take a look at some specific examples.
+
+| Day | Event    | distinct_id              | `person_id` |
+| --- | -------- | ------------------------ | ----------- |
+| 1   | other    | Alice                    | user-1      |
+| 2   | pageview | anon-1                   | user-2      |
+| 2   | identify | Alice (anon-id = anon-1) | user-1      |
+
+In this case, we have a user Alice who sends an 'other' event on day `1` from her mobile phone.
+On day 2, Alice decides to view the homepage from her desktop where she isn't logged in. This results in the pageview event being associated with a newly created Person (`user-2`).
+She then logs in to her account, which sends an identify event that merges `user-2` into `user-1`.
+This mean that we delete `user-2` from the persons table and all future events from `anon-1` will be tied to `user-1` (note that we never alter the events table to reflect this).
+
+In this case, we’d show 1 unique user in the trend graph for pageviews, but since `user-2` was deleted during the merge, we would show 0 users in the person modal.
```

Comment:
is person modal Postgres-based?

Reply:
we just drop deleted ones - CH also has deleted persons